bot/cogs/weather_cog.py

```diff
@@ -0,0 +1,206 @@
+                if (response.status == 200):
+                    res_geo_json = await response.json()
+                    city = res_geo_json.get('standard',{}).get('city',{})
+                    lon = res_geo_json.get('longt',{})
+                    lat = res_geo_json.get('latt',{})
+                    
+                    url = "https://api.openweathermap.org/data/2.5/onecall"
+
+                    queryparams = {
+                        'lat' : lat,
+                        'lon' : lon,
+                        'exclude' : 'minutely,hourly',
+                        'appid' : self.weather_api_key,
+                        'units' : 'imperial',
+                        'lang' : 'en'
+                        }
+
+                    weatherPages = []
+
+                    async with aiohttp.request("GET", url, params=queryparams) as response:
+                        if (response.status == 200):
+                            res_json = await response.json()
+                            weatherPages = self.getPageData(lat,lon,res_json,city)
+                        
+                        else:
+                            embed = discord.Embed(title='OpenWeatherMap Weather', color=Colors.Error)
+                            ErrMsg = f'Error Code: {response.status}'
+                            embed.add_field(name='Error with weather API', value=ErrMsg, inline=False)
+                            await ctx.send(embed=embed)
+                            return
+
+                        await self.bot.messenger.publish(Events.on_set_pageable,
+                            embed_name = 'OpenWeatherMap Weather',
+                            field_title = 'Current Conditions and 7-Day Forecast',
+                            pages = weatherPages,
+                            author = ctx.author,
+                            channel = ctx.channel)
+                
+                else:
+                    embed = discord.Embed(title='OpenWeatherMap Weather', color=Colors.Error)
+                    ErrMsg = f'Error Code: {response.status}'
+                    embed.add_field(name='Error with geocode API', value=ErrMsg, inline=False)
+                    await ctx.send(embed=embed)
+                    return
+                """ 
+                # FOR DEBUGGING. Takes a minute or two for message to display when enabled
+                await self.bot.messenger.publish(Events.on_set_pageable,
+                    embed_name = 'Geocoding Results',
+                    field_title = 'Location',
+                    pages = [f'Query: {loc}\n\nLatitude: {lat}\nLongitude: {lon}'],
+                    author = ctx.author,
+                    channel = ctx.channel)
+                """
+        except Exception as err:
+            err_str = str(err)
+            err_str = re.sub(self.geocode_api_key, "CLASSIFIED", err_str)
+            err_str = re.sub(self.weather_api_key, "CLASSIFIED", err_str)
```

Comment:
im glad we found this vulnerability, its obscure as heck but its always good to cover our bases

Reply:
Props to @CodysGithub for finding it

Reply:
What exactly is this vulnerability?

Reply:
it is a vulnerability utilizing the robots.txt on a website. The general idea is that the user would do something like `!weather ..\..\..\robots.txt` and then the cog would try to open the file because it exists. But, because it isn't a JSON, it would raise an error where the cog returns the URL of the file.

Since the weather and define APIs have their keys in the URL, this will give the end-user the URL. The user could then use that for their own malicious purposes, potentially getting our key banned or spamming it with requests.

Reply:
Oh interesting. Good find